src/main/java/dataloader/CsvDataLoader.java

```diff
@@ -25,24 +31,92 @@
 
     public static class MapLoader {
 
+
+        /**
+         * Create a NumericsSample from the given column of the
+         * in the TreeMap
+         */
+        public static NumericSample buildSample(TreeMap<String, List> dataSet, String colName){
+
+            if(dataSet == null){
+
+                throw new IllegalArgumentException("Null data set given");
+            }
+
+            NumericSample sample = null;
+
+            if(!dataSet.containsKey(colName)){
+
+                if(Configuration.ENABLE_WARNINGS) {
+                    System.out.println("WARNING: Column " + colName + " not in dataset");
+                }
+
+                return new NumericSample(colName, 0);
+            }
+
+            List<Double> data = ParseUtils.parseAsDouble( dataSet.get(colName) );
+            sample = new NumericSample(colName, data, false);
+            return sample;
+        }
+
         /**
          * Simple method that parses data set from a csv file
-         * without knowing the headers of the file
+         * The CSV file should NOT have the last column ending with comma
          */
-        public static TreeMap parseFile(File csvFile) throws IOException {
```

Comment:
check out this implantation `/**
         * Simple method that parses a
         * data set from a csv file
         * without headers
         */
        public static HashMap parseFile(File csvFile) throws IOException {

            Reader csvData = new FileReader(csvFile);
            HashMap<String, Map<String, String>> dataSet = new HashMap();

            CSVParser parser = CSVParser.parse(csvData, CSVFormat.DEFAULT);

            for (CSVRecord record : parser) {
                dataSet.put(record.toString(), record.toMap());
            }
            return dataSet;
        }`
for here [this commit](https://github.com/jethronap/JavaDataAnalysis/commit/f36f01119e851c046a38787b9a8c51d6d4800463).

sorry for the irrelevant commit inside that branch. 
Tell what you think.

Reply:
What is this meant to do?

Reply:
this creates a hash map where the we have key value pairs. the keys are the record numbers and the keys are the values of the columns in strings. pull it and run debug the test and you'll understand what this does exactly.

Reply:
this is what you get: ```CSVRecord [comment='null', recordNumber=3, values=[4,  5,  6]] -> {LinkedHashMap@1746} size = 0``` And I don't think this is what you want....

Reply:
The ```CSVParser``` reads rows....the ```HashMap``` should store ```ColName->list of values```. In order to get this list of values you must iterated over all rows and extract the value corresponding to the column you want....

Reply:
Anyways, I agree we should use sth like ```HashMap``` will change it in this branch

Reply:
but the `LinkedHashMap` of the overall CsvRecord gives a `size = 3`. in any case I will deep a little bit more. I will continue along these lines, i.e. parse a csv without having a header row. the opposite is covered by the tablesaw implementation.

Reply:
I think the documentation of the Maploader class need to be changed from TreeMap to HashMap or plain Map